cloud/aws/beanstalk/monitors-beanstalk.tf

```diff
@@ -0,0 +1,136 @@
+		disabled              = coalesce(var.latency_p90_disabled_warning, var.latency_p90_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.latency_p90_notifications_warning, var.latency_p90_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "5xx_error_rate" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS beanstalk 5xx error rate"
+
+	program_text = <<-EOF
+		A = data('ApplicationRequests5xx', filter=filter('namespace', 'AWS/ElasticBeanstalk') and filter('stat', 'sum') and filter('InstanceId', '*') and ${module.filter-tags.filter_custom}){var.5xx_error_rate_aggregation_function}
+		B = data('ApplicationRequestsTotal', filter=filter('namespace', 'AWS/ElasticBeanstalk') and filter('stat', 'sum') and filter('InstanceId', '*') and ${module.filter-tags.filter_custom}){var.5xx_error_rate_aggregation_function}
+		signal = (A/B).scale(100).${var.5xx_error_rate_transformation_function}(over='${var.5xx_error_rate_transformation_window}')
+		detect(when(signal > ${var.5xx_error_rate_threshold_critical})).publish('CRIT')
+		detect(when(signal > ${var.5xx_error_rate_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high > ${var.5xx_error_rate_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.5xx_error_rate_disabled_critical, var.5xx_error_rate_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.5xx_error_rate_notifications_critical, var.5xx_error_rate_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.5xx_error_rate_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.5xx_error_rate_disabled_warning, var.5xx_error_rate_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.5xx_error_rate_notifications_warning, var.5xx_error_rate_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "root_filesystem_usage" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Beanstalk Instance root file system usage"
+
+	program_text = <<-EOF
+		signal = data('RootFilesystemUtil', filter=filter('namespace', 'AWS/ElasticBeanstalk') and filter('stat', 'lower') and filter('InstanceId', '*') and ${module.filter-tags.filter_custom}){var.root_filesystem_usage_aggregation_function}.${var.root_filesystem_usage_transformation_function}(over='${var.root_filesystem_usage_transformation_window}')
```

Comment:
stat should be upper no?

Reply:
upper and lower is valid.  Its more sensitive if we alert when lower value is above threshold